packages/react/src/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -1,33 +1,356 @@
+  )
+
+  useResizeObserver(handleResize, containerRef)
+
+  useEffect(() => {
+    if (
+      overflowMenuEnabled &&
+      (overflow === 'menu' || overflow === 'menu-with-root') &&
+      childArray.length > 5 &&
+      menuItems.length === 0
+    ) {
+      const containerWidth = containerRef.current?.offsetWidth || 800
+      const result = calculateOverflow(containerWidth)
+      setVisibleItems(result.visibleItems)
+      setMenuItems(result.menuItems)
+      setEffectiveHideRoot(result.effectiveHideRoot)
+    }
+  }, [overflow, childArray, calculateOverflow, menuItems.length, overflowMenuEnabled])
+
+  const finalChildren = React.useMemo(() => {
+    if (overflowMenuEnabled) {
+      if (overflow === 'wrap' || menuItems.length === 0) {
+        return React.Children.map(children, child => <li className={classes.ItemWrapper}>{child}</li>)
+      }
+
+      let effectiveMenuItems = [...menuItems]
+      // In 'menu-with-root' mode, include the root item inside the menu even if it's visible in the breadcrumbs
+      if (!effectiveHideRoot) {
+        effectiveMenuItems = [...menuItems.slice(1)]
+      }
+      const menuElement = (
+        <li className={classes.BreadcrumbsItem} key="breadcrumbs-menu">
+          <BreadcrumbsMenuItem
+            ref={measureMenuButton}
+            items={effectiveMenuItems}
+            aria-label={`${effectiveMenuItems.length} more breadcrumb items`}
+          />
+          <ItemSeparator />
+        </li>
+      )
+
+      const visibleElements = visibleItems.map((child, index) => (
+        <li className={classes.BreadcrumbsItem} key={`visible + ${index}`}>
+          {child}
+          <ItemSeparator />
+        </li>
+      ))
+
+      const rootElement = (
+        <li className={classes.BreadcrumbsItem} key={`rootElement`}>
+          {rootItem}
+          <ItemSeparator />
+        </li>
+      )
+
+      if (effectiveHideRoot) {
+        // Show: [overflow menu, leaf breadcrumb]
+        return [menuElement, ...visibleElements]
+      } else {
+        // Show: [root breadcrumb, overflow menu, leaf breadcrumb]
+        return [rootElement, menuElement, ...visibleElements]
+      }
+    }
+  }, [overflowMenuEnabled, overflow, menuItems, effectiveHideRoot, measureMenuButton, visibleItems, rootItem, children])
+
+  return overflowMenuEnabled ? (
+    <BoxWithFallback
+      as="nav"
+      className={clsx(className, classes.BreadcrumbsBase)}
+      aria-label="Breadcrumbs"
+      sx={sxProp}
+      ref={containerRef}
+      data-overflow={overflow}
+    >
+      <BreadcrumbsList>{finalChildren}</BreadcrumbsList>
+    </BoxWithFallback>
+  ) : (
     <BoxWithFallback as="nav" className={clsx(className, classes.BreadcrumbsBase)} aria-label="Breadcrumbs" sx={sxProp}>
       <BreadcrumbsList>{wrappedChildren}</BreadcrumbsList>
     </BoxWithFallback>
   )
 }
 
+const ItemSeparator = () => {
+  return (
+    <span className={classes.ItemSeparator}>
```

Comment:
You might need some labeling on this for a11y support (just a hunch, would look it up)

Reply:
It's not focusable. I'm not sure if the label is required. What would you expect in the label?
cc - @janmaarten-a11y